.github/workflows/integrate.yaml

```
@@ -33,18 +37,26 @@ jobs:
 
   unit:
     name: Unit Tests
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v3
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v5.3.0
```

nitpick: newest action version
```suggestion
        uses: actions/setup-python@v5.4.0
```